### PR TITLE
Latent metric anomaly detection

### PR DESCRIPTION
**Pull request by ndeutschmann**
This was work in most part performed by @fcraighero while at IBM Research Zurich. It implements three algorithms for epistemic uncertainty estimation/anomaly detection based on proxies for data density in the latent space of neural networks.

It provides the following new posthocUQ methods:

- `uq360.algorithms.layer_scoring.knn.KNN`
- `uq360.algorithms.layer_scoring.aklpe.AKLPE`
- `uq360.algorithms.layer_scoring.mahalanobis.Mahalanobis`

The first two methods are based on k-Nearest-Neighbor searches, for which we provide three solutions
- `uq360.utils.transformers.nearest_neighbors.exact.ExactNearestNeighbors` based on `sklearn`
- `uq360.utils.transformers.nearest_neighbors.pynndescent.PyNNDNearestNeighbors` based on `pynndescent`
- `uq360.utils.transformers.nearest_neighbors.faiss.FAISSNearestNeighbors` based on `faiss`

Still TODO: implementing tests (code has been checked of course though). Opening this PR to allow discussions.

### CODE REVIEW

**Comment by pronics2004** [comment]
This is looking great!

The travis build failures should be fixed after merging the latest `main` branch.

Also, when you add tests, you could add the additional dependencies to `.travis.yml` as a temporary fix so that travis can run those tests.

**Comment by ndeutschmann** [comment]
I opened #25 with signed-off commits  but the same content. I'm closing this
